sqlalchemy-stubs/sql/ddl.pyi

```diff
@@ -1,68 +1,88 @@
 from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import List
 from typing import Optional
-
-from . import roles as roles
-from .base import Executable as Executable
-from .base import SchemaVisitor as SchemaVisitor
-from .elements import ClauseElement as ClauseElement
-from .. import exc as exc
-from .. import util as util
-from ..util import topological as topological
+from typing import Sequence
+from typing import Set
+from typing import Tuple
+from typing import TypeVar
+from typing import Union
+
+from . import roles
+from .base import Executable
+from .base import SchemaVisitor
+from .elements import ClauseElement
+from .schema import ForeignKey
+from .schema import ForeignKeyConstraint
+from .schema import Index
+from .schema import SchemaItem
+from .schema import Table
+from ..engine import Connection
+from ..engine import Dialect
+from ..engine import Engine
+
+_DDLE = TypeVar("_DDLE", bound=DDLElement)
 
 class _DDLCompiles(ClauseElement): ...
 
 class DDLElement(roles.DDLRole, Executable, _DDLCompiles):
-    target: Any = ...
+    target: Optional[SchemaItem] = ...
     on: Any = ...
-    dialect: Any = ...
+    dialect: Union[str, Tuple[str, ...], List[str], Set[str]] = ...
     callable_: Any = ...
-    def execute(
-        self, bind: Optional[Any] = ..., target: Optional[Any] = ...
-    ): ...
-    def against(self, target: Any) -> None: ...
     state: Any = ...
-    def execute_if(
+    bind: Optional[Union[Engine, Connection]] = ...
+    def execute(  # type: ignore[override]
         self,
-        dialect: Optional[Any] = ...,
+        bind: Optional[Union[Engine, Connection]] = ...,
+        target: Optional[SchemaItem] = ...,
+    ) -> Any: ...
+    def against(self: _DDLE, target: Optional[SchemaItem]) -> _DDLE: ...
+    def execute_if(
+        self: _DDLE,
+        dialect: Optional[
+            Union[str, Tuple[str, ...], List[str], Set[str]]
+        ] = ...,
         callable_: Optional[Any] = ...,
         state: Optional[Any] = ...,
-    ) -> None: ...
-    def __call__(self, target: Any, bind: Any, **kw: Any): ...
-    def bind(self): ...
-    bind: Any = ...
+    ) -> _DDLE: ...
+    def __call__(
+        self, target: SchemaItem, bind: Union[Engine, Connection], **kw: Any
+    ) -> Any: ...
 
 class DDL(DDLElement):
     __visit_name__: str = ...
-    statement: Any = ...
+    statement: str = ...
     context: Any = ...
     def __init__(
         self,
-        statement: Any,
-        context: Optional[Any] = ...,
-        bind: Optional[Any] = ...,
+        statement: str,
+        context: Optional[Dict[str, Any]] = ...,
+        bind: Optional[Union[Engine, Connection]] = ...,
     ) -> None: ...
 
 class _CreateDropBase(DDLElement):
```

Comment:
I guess this could be generic on the `element` type, but it's probably not worth it